### ISSUE TITLE
[Links]: Generating content for a download

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/vets-design-system-documentation#1799]
### Duplicate check

- [X] I've searched for any related issues and avoided creating a duplicate issue.

### This update is for:

Component

### What is the name?

Link

### What is the nature of this update?

- [ ] How to build this component/pattern
- [ ] When to use this component/pattern
- [X] When to use something else
- [X] Usability guidance
- [X] Accessibility
- [X] Implementation
- [ ] Research insights
- [ ] Package information
- [ ] Addition to Forms Library documentation
- [ ] Update to existing Forms Library documentation

## Download links should have guidance on generating content scenarios

Recently in an audit on letters, we encountered buttons that _download_ content. At first glance, according to design system guidance (and accessibility standards), we should be using a download link. However, in this instance, the content being downloaded does not exist within an existing location to point to via URL; rather, they are being generated real time (hence an action) before providing the option to download.

More discovery could be used by an a11y specialist to explore how to handle this scenario, and if more upstream recommendations may exist (e.g. pre-generating all downloadable content on a page so they can be offered as download links for consistency?

### Download button needs review (Moderate)
If the user is downloading pdfs that have already been generated, they should be styled and [written as links following design system guidance](https://design.va.gov/components/link/#download).

If the user is _generating_ a pdf, and then downloading it, we should investigate if we can pre-generate the pdfs (and their variations) so that they may be styled as links. Otherwise we may have to style and code them as buttons.

We should try our best to use links (instead of buttons) are preferred as they can provide the user additional context and actions including:
- opening in the browser or in a new tab
- a count of pages
- the type of document it is (PDF, word doc, etc)
- how large it is (for users with low bandwidth connections or limited data)

https://user-images.githubusercontent.com/14154792/184921694-3315ebc4-e781-447b-a942-357839887299.MP4

## Reproduction steps

1. [Log in as user 299 and get to step 2](https://staging.va.gov/records/download-va-letters/letters/letter-list)
2. Open the benefit summary accordion

**Comment by jeana-adhoc** [comment]
This has been addressed in many ways!

- [We have a download link variant now](https://design.va.gov/components/link/#download)
- We have guidance on when to use a [button or a link](https://design.va.gov/components/link/#choose-the-right-element-buttons-vs-links)
